service/routes/rum.js

```diff
@@ -0,0 +1,125 @@
+/* Endpoints for Real User Monitoring of detect and perf results */
+
+'use strict';
+
+const polyfillio = require('../../lib');
+const metrics = require('../metrics');
+const express = require('express');
+
+let mysql;
+if (process.env.MYSQL_DSN) {
+	require('promise-mysql').createConnection(process.env.MYSQL_DSN).then(conn => {
+		mysql = conn;
+	});
+}
+
+const router = express.Router();  // eslint-disable-line new-cap
+const blankGif = new Buffer([0x47, 0x49, 0x46, 0x38, 0x39, 0x61, 0x01, 0x00, 0x01, 0x00, 0x80, 0x00, 0x00, 0xff, 0xff, 0xff, 0x00, 0x00, 0x00, 0x2c, 0x00, 0x00, 0x00, 0x00, 0x01, 0x00, 0x01, 0x00, 0x00, 0x02, 0x02, 0x44, 0x01, 0x00, 0x3b]);
+
+
+/**
+ * GET /v2/recordData
+ *  - <featureName>: 1 if it passed the detect, else 0
+ *  - dns: perf.domainLookupEnd - perf.domainLookupStart
+ *  - connect: perf.connectEnd - perf.connectStart
+ *  - req: perf.responseStart - perf.requestStart
+ *  - resp: perf.responseEnd - perf.responseStart
+ */
+router.get('/v2/recordRumData', (req, res) => {
+
+	res.set('Cache-Control', 'no-store');
+
+	const logquery = 'INSERT INTO requests SET perf_dns=?, perf_connect=?, perf_req=?, perf_resp=?, ua_family=?, ua_version=?, lat=?, lng=?, country=?, data_center=?, refer_domain=?';
+	const referer = (req.get('referer') || '').replace(/^(https?:\/\/)?(www\.)?(.+?)(\:\d+)?([\/\?].*)?$/, '$3');
+	const ua = polyfillio.normalizeUserAgent(req.get('user-agent')).split('/');
+	const logdata = [req.query.dns, req.query.connect, req.query.req, req.query.resp, ua[0], ua[1], req.get('geo-lat'), req.get('geo-lng'), req.get('geo-country'), req.get('data-center'), referer];
```

Comment:
Yes, it works when I run it locally.  The DC and geoip stats just get set to null.